### DISCUSSION TITLE
Is there a way to disable references from another extension?

### DISCUSSION THREAD

**Discussion by matheusccastroo** [microsoft/vscode-discussions#360] [Extension Development QnA] [answered]
Hello!

I'm having an issue that I guess it's a conflict between my extension and the built-in typescript/javascript extension references/definition resolver.

Basically my issue is the following: let's say I have the token `links.insert`. This token is a method from an object and it has only one reference on the code. When I `ctrl + left click` on it, it should request the references for this token (because on the definition resolver I return the same location as the request), and as we have only one reference, it should go to it "automatically".
However, the problem is that the built-in extension is bringing a "reference" that should not be there. Check the image below:
![Screenshot_20230102_035555](https://user-images.githubusercontent.com/48069682/210202038-daa118ae-8477-4966-a02d-4de705ad7dd6.png)

The `info.js` is the correct reference, but the one that is shown is the one the user is at already.

Other example is the following: suppose I have the token `Template.info.onRendered` and I `ctrl + left click` on the `info`. What should happen is that I would request the definition and as I'm also returning for this case the same location, it should trigger the references request. The problem is that this isn't happening and it's actually triggering the definition request.

Check the image below:
![Screenshot_20230102_035533](https://user-images.githubusercontent.com/48069682/210202374-b6d9fb3c-39bb-4c0c-886a-c1c044dc541d.png)

For this case, as it's just doing the definition request, it's not bringing the references. I think the reason for this is because we have the types file for this interface, which makes sense, but is there a way to disable this?

When I disable the `TypeScript and JavaScript Language Features` extension, everything work as expected. Any ideas of what I'm doing wrong? Is there a way to disable this specific features from this built-in extension? Or maybe a way to force my extension references to go through?

**Comment by r3m0t** [comment, marked as answer]
You can try writing a typescript server plugin to identify and remove these references, or to move your code to find references into the typescript server so that it tries your version first and uses the standard reference code as a fallback. https://code.visualstudio.com/api/references/contribution-points#contributes.typescriptServerPlugins

Maybe there's another way.

**Reply by matheusccastroo** [reply]
@r3m0t Oh, I didn't know about server plugins, that seems like the way to go. Thank you!

**Reply by matheusccastroo** [reply]
Giving an update here: it worked!

With the plugin I was able to filter the completion items, definitions and references the TypeScript server creates and filter it based on the node.

Thanks for the help!
